crates/bevy_reflect/src/serde/deserialize_reflect.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{FromType, Reflect, TypeRegistry};
+use serde::{Deserialize, Deserializer};
+
+/// Trait used to provide finer control when deserializing a reflected type with one of
+/// the reflection deserializers.
+///
+/// This trait is the reflection equivalent of `serde`'s [`Deserialize`] trait.
+/// The main difference is that this trait provides access to the [`TypeRegistry`],
+/// which means that we can use the registry and all its stored type information
+/// to deserialize our type.
```

Comment:
Maybe add a note about the ability to deserialize a type by name? It's kind of obvious, but is probably the most common use of the type registry here, no?

Reply:
That's already achievable with the `TypedReflectDeserializer` API. The main benefit this provides is avoiding boilerplate caused by using `DeserializeSeed`. Maybe I should include that?

Reply:
I think an example would be nice. I just barely follow that comment with all the context from the past 12 hours of discussion. This whole reflection area is quite hard to get ramped up to.

Reply:
I'm hesitant to put an example in since it could easily become quite complex, especially due to the serde logic. I could maybe add a condensed version with the logic commented out? But I feel like that doesn't really give the full picture either.

Reply:
Let's close for now, can be added later if needed.